mne/tests/test_annotations.py

```diff
@@ -125,4 +134,82 @@ def test_raw_reject():
+    assert_allclose(raws_zero[0][0], raws_stop_concat[0][0], atol=1e-14)
+    # ensure that our Annotations cut up the filtering properly
+    raws_concat_pass = raws_concat.copy().filter(skip_by_annotation='edge',
+                                                 **kwargs_pass)
+    assert_allclose(raws_concat[0][0], raws_concat_pass[0][0], atol=1e-14)
+    raws_concat_stop = raws_concat.copy().filter(skip_by_annotation='edge',
+                                                 **kwargs_stop)
+    assert_allclose(raws_zero[0][0], raws_concat_stop[0][0], atol=1e-14)
+    # one last test: let's cut out a section entirely:
+    # here the 1-3 second window should be skipped
+    raw = raws_concat.copy()
+    raw.annotations.append(1., 2., 'foo')
+    raw.filter(l_freq=50., h_freq=None, fir_design='firwin',
+               skip_by_annotation='foo')
+    # our filter will zero out anything not skipped:
+    mask = np.concatenate((np.zeros(1000), np.ones(2000), np.zeros(1000)))
+    expected_data = raws_concat[0][0][0] * mask
+    assert_allclose(raw[0][0][0], expected_data, atol=1e-14)
+
+
+def test_annotation_omit():
+    """Test raw.get_data with annotations."""
+    data = np.concatenate([np.ones((1, 1000)), 2 * np.ones((1, 1000))], -1)
+    info = create_info(1, 1000., 'eeg')
+    raw = RawArray(data, info)
+    raw.annotations = Annotations([0.5], [1], ['bad'])
+    expected = raw[0][0]
+    assert_allclose(raw.get_data(reject_by_annotation=None), expected)
+    # nan
+    expected[0, 500:1500] = np.nan
+    assert_allclose(raw.get_data(reject_by_annotation='nan'), expected)
+    got = np.concatenate([raw.get_data(start=start, stop=stop,
+                                       reject_by_annotation='nan')
+                          for start, stop in ((0, 1000), (1000, 2000))], -1)
+    assert_allclose(got, expected)
+    # omit
+    expected = expected[:, np.isfinite(expected[0])]
+    assert_allclose(raw.get_data(reject_by_annotation='omit'), expected)
+    got = np.concatenate([raw.get_data(start=start, stop=stop,
+                                       reject_by_annotation='omit')
+                          for start, stop in ((0, 1000), (1000, 2000))], -1)
+    assert_allclose(got, expected)
+    assert_raises(ValueError, raw.get_data, reject_by_annotation='foo')
+
+
+def test_annotation_epoching():
+    """Test that annotations work properly with concatenated edges."""
+    # Create data with just a DC component
+    data = np.ones((1, 1000))
+    info = create_info(1, 1000., 'eeg')
+    raw = concatenate_raws([RawArray(data, info) for ii in range(3)])
+    events = np.array([[a, 0, 1] for a in [0, 500, 1000, 1500, 2000]])
+    epochs = Epochs(raw, events, tmin=0, tmax=0.999, baseline=None,
+                    preload=True)  # 1000 samples long
+    assert_equal(len(epochs.drop_log), len(events))
+    assert_equal(len(epochs), 3)
+    assert_equal([0, 2, 4], epochs.selection)
```

Comment:
ok, this suggests that edge annotations are neither the first sample of the added segment nor the last sample nor last one of the segment previous to the added one OR that I don't get how 'BAD boundary' affects epoching when it is the first or last sample of an epoch...